operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -554,6 +608,108 @@ def validateServerTemplateNapListenPortIsSet(self, server_or_template):
+    if migrationPolicy is not None and migrationPolicy != "manual":
+      leasingRequired = True
+      if self.isConsensusLeasing(cluster):
+        if isServerTemplate:
+          type_str = "server template "
+        else:
+          type_str = "server "
+        self.addConsensusLeasingError("JTA migratable target in " + type_str + self.name(server_or_template)
+                                      + " in cluster " + self.name(cluster) + " requires automatic service migration")
+    return leasingRequired
+
+  def addMigrationError(self, message):
+    self.addError(message + " Set SKIP_LEASING_VALIDATIONS environment variable to 'True' to skip this validation.")
+
+  def addConsensusLeasingError(self, message):
+    self.addError(message + ", but cluster is configured to use consensus leasing which is not " +
+                  "supported by the WebLogic Kubernetes Operator. " +
+                  "Please configure cluster to use database leasing " +
+                  "or set SKIP_LEASING_VALIDATIONS environment variable to 'True' to skip this validation.")
+
+  def findMigratableTarget(self, name):
+    for migratableTarget in self.env.getDomain().getMigratableTargets():
+      if migratableTarget.getName() == name:
+        return migratableTarget
+    return None
+
+  def validateMigratableTargets(self, cluster):
+    leasingRequired = False;
+    for migratableTarget in self.env.getDomain().getMigratableTargets():
+      if migratableTarget.getCluster() is cluster:
+        # possible migration policy values are "manual", "exactly-once", and "failure-recovery".
+        # All values except "manual" requires cluster leasing.
+        migrationPolicy = migratableTarget.getMigrationPolicy()
+        if migrationPolicy is not None and migrationPolicy != "manual":
+          # check JMS resources targeted to this migratable target
+          leasingRequired |= self.validateJMSResourcesLeasing(migratableTarget)
+    return leasingRequired
+
+  def validateJMSResourcesLeasing(self, migratableTarget):
+    leasingRequired = False;
+    domain = self.env.getDomain()
+    for messagingBridge in domain.getMessagingBridges():
+      leasingRequired |= self.validateJMSResourceMigrationPolicy(messagingBridge, migratableTarget, "Messaging bridge")
+    for jdbcStore in domain.getJDBCStores():
+      leasingRequired |= self.validateJMSResourceMigrationPolicy(jdbcStore, migratableTarget, "JDBCStore")
+    for fileStore in domain.getFileStores():
+      leasingRequired |= self.validateJMSResourceMigrationPolicy(fileStore, migratableTarget, "FileStore")
+    return leasingRequired
+
+  def validateJMSResourceMigrationPolicy(self, jmsResource, migratableTarget, typeStr):
+    leasingRequired = False
+    # valid migration policy values are "Off", "On-Failure", and "Always".
+    # Cluster leasing must be configured for "On-Failure" and "Always".
+    if jmsResource.getMigrationPolicy() != None and jmsResource.getMigrationPolicy() != "Off":
```

Comment:
I'm not sure the logic is quite right here.

- The MigrationPolicy attribute directly on stores, path services, and bridges cluster targeted, and leasing, should only be checked when these resource are cluster targeted.  Leasing is needed when when this is not "Off".

- The MigrationPolicy on MTs for the JMS case only apply to jms servers, saf agents, stores, and bridges, that are migratable targeted, and only need consensus leasing when the MT migration policy is not "Manual".

The two cases should be treated distinctly - not as an overlapping union.

Reply:
Updated and added check for cluster targeted